pkg/controller/operators/olm/overrides/inject_test.go

```diff
@@ -746,6 +746,30 @@ func TestInjectNodeSelectorIntoDeployment(t *testing.T) {
 				NodeSelector: map[string]string{"foo": "bar"},
 			},
 		},
+		{
+			// Existing PodSpec is left alone if nodeSelector is nil
+			// Expected: PodSpec is not changed
+			name: "WithNilNodeSelector",
+			podSpec: &corev1.PodSpec{
+				NodeSelector: defaultNodeSelector,
+			},
+			nodeSelector: nil,
+			expected: &corev1.PodSpec{
+				NodeSelector: defaultNodeSelector,
+			},
+		},
+		{
+			// Existing PodSpec is set to an empty map if the nodeSelector is an empty map
```

Comment:
how would this be serialized?

Reply:
ah, forgetting YAML is a superset of JSON...

```yaml
emptyMap: {}
```

Reply:
I don't believe it can.